Automattic/wp-calypso
# FSE: Scaffold editor-domain-picker package

#### Changes proposed in this Pull Request

Scaffold a "editor-domain-picker" package:
- package scripts for building
- PHP enqueuing
- "Feature flag" (`A8C_FSE_DOMAIN_PICKER_ENABLE`)
- Some friendly placeholder JS to show the bundling is working.

#### Testing instructions

Apply D44014-code

Try with the following conditions (add to sandbox):

- Nothing special defined (feature disabled)
- `define( 'A8C_FSE_DOMAIN_PICKER_ENABLE', false );` (feature disabled)
- `define( 'A8C_FSE_DOMAIN_PICKER_ENABLE', true );` (feature enabled)

1. Sandbox a simple site and load the editor: you should see the friendly debugging messages on the console if the constant enables the feature.
1. When not enabled, the script should not be enqueued, check for it in the network tab. `editor-domain-picker` and shared dependency `common-data-stores`.
1. For other pages (not the block editor) nothing should be enqueued or run. Verify same behavior as when the feature is disabled.

No new/related PHP errors or warnings should be generated under any of the above conditions.
